plugins/kubernetes/test/models/kubernetes/api/horizontal_pod_autoscaler_test.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+require_relative '../../../test_helper'
+
+SingleCov.covered!
+
+describe Kubernetes::Api::HorizontalPodAutoscaler do
+  let(:start_time) { "2017-03-31T22:56:20Z" }
+  let(:event) { {metadata: {creationTimestamp: start_time}, type: 'Normal'} }
+  let(:events) { [event] }
+  let(:hpa) { Kubernetes::Api::HorizontalPodAutoscaler.new }
+
+  describe "HorizontalPodAutoscaler with failures we don't care about" do
+    before do
+      event.merge!(kind: 'HorizontalPodAutoscaler', type: 'Warning')
+    end
+
+    it "ignores failing to get metrics" do
+      event[:reason] = 'FailedGetMetrics'
+
+      refute hpa.events_indicating_failure(events).empty?, 'FailedGetMetrics must not be recognized as failures'
+    end
+
+    it "ingores failures to scale" do
+      event[:reason] = 'FailedRescale'
+
+      refute hpa.events_indicating_failure(events).empty?, 'FailedRescale must not be recognized as failures'
+    end
+
+    it "ignores failing to get resource metrics" do
+      event[:reason] = 'FailedGetResourceMetric'
+
+      refute hpa.events_indicating_failure(events).empty?, 'FailedGetResourceMetric must not be recognized as failures'
+    end
+
+    it "ignores failing to get external metrics" do
+      event[:reason] = 'FailedGetExternalMetric'
+
+      refute hpa.events_indicating_failure(events).empty?, 'FailedGetExternalMetric must not be recognized as failures'
```

Comment:
I'd recommend leaving out these extra messages since if something goes wrong usually we have to dig in anyway and they get out of date quickly :shrug:
```suggestion
      refute hpa.events_indicating_failure(events).empty?
```